src/engine/TransitivePathHashMap.h

```diff
@@ -12,7 +12,28 @@
 #include "engine/TransitivePathImpl.h"
 #include "engine/idTable/IdTable.h"
 
-class TransitivePathHashMap : public TransitivePathImpl<Map> {
+struct HashMapWrapper {
+  Map map_;
+
+  auto successors(const Id node) const {
+    auto iterator = map_.find(node);
+    if (iterator == map_.end()) {
+      return std::vector<Id>();
+    }
+    std::vector<Id> result(iterator->second.begin(), iterator->second.end());
+    return result;
+  }
+};
```

Comment:
Don't copy to a vector, but return a `const Set&`. 
You can make an empty set another member of the `HashMapWrapper`, so that you have something to return in the case of `not found`.

Reply:
I solved this, but I was not sure where to get the `ad_utility::AllocatorWithLimit<Id>`. There may be a smarter solution.